gcip/addons/container/jobs/crane.py

```diff
@@ -118,3 +118,65 @@ def push(
         job.prepend_scripts(*docker_client_config.get_shell_command())
 
     return job
+
+
+def pull(
+    src_registry: Union[Registry, str],
+    *,
+    tar_path: Optional[str] = None,
+    image_name: Optional[str] = None,
+    image_tag: Optional[str] = None,
+    docker_client_config: Optional[DockerClientConfig] = None,
+    crane_image: Optional[Union[Image, str]] = None,
+) -> Job:
+    """
+    Creates a job to pull container image from remote container registry with `crane`.
+
+    Args:
+        src_registry (str): Registry URL to pull container image from.
+        image_name (str): Container image with namespace to pull from `src_registry`.
+        image_tag (str): Tag of the image which will be pulled.
+        tar_path (Optional[str], optional): Path where to save the container image tarball.
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_DIR`. Defaults to None.
+        docker_client_config (Optional[DockerClientConfig], optional): Creates the Docker configuration file base on objects settings,
+            to authenticate against given registries. Defaults to a `DockerClientConfig` with login to the official Docker Hub
+            and expecting credentials given as environment variables `REGISTRY_USER` and `REGISTRY_LOGIN`.
+        crane_image (Optional[Union[Image, str]], optional): Container image which contains `crane` command.
+            Defaults to PredefindedImages.CRANE.
+
+    Returns:
+        Job: Returns a `gcip.Job`, with neccessary configuration to pull a container image from a remote registry and stores it in `tar_path`.
+            Job runs in ```stage=pull```
+    """
+    if not crane_image:
+        crane_image = PredefinedImages.CRANE
+
+    if not tar_path:
+        tar_path = PredefinedVariables.CI_PROJECT_DIR
+
+    if not image_name:
+        image_name = PredefinedVariables.CI_PROJECT_NAME
+    image_path = image_name.replace("/", "_")
+
+    if not docker_client_config:
+        docker_client_config = DockerClientConfig()
+        docker_client_config.add_auth(registry=Registry.DOCKER)
+
+    if not image_tag:
+        if PredefinedVariables.CI_COMMIT_TAG:
+            image_tag = PredefinedVariables.CI_COMMIT_TAG
+        else:
+            image_tag = PredefinedVariables.CI_COMMIT_REF_SLUG
+
+    job = Job(
```

Comment:
Is that `pull` job usable without having set the jobs `cache` ( or `artifacts`)?

Reply:
Mhh. I think you refer to #59? I didn't want to implement it in this PR. 
This issue is dedicated to all container jobs, therefore i think the right way would be to implement it in another PR?

Reply:
Ah, nice - I haven't thought about #59 . Yes and no.

Yes - this relates to #59, as the user of the gcip doesn't need to bother with the cache configuration of this job.

No - this is not dedicated to all container jobs. My proposal is to configure the `crane.pull()` job with a default cache. However no other job has to use that cache.

The cool thinkg is, that one of your last commits already solved the main show stopper of #59 - jobs would not expose their cache as property:
https://github.com/dbsystel/gitlab-ci-python-library/blob/081b33edd2f995ab3b869ed176926d5509e23a27/gcip/core/job.py#L234

That means it is quite easy to simply add a default cache here, which other jobs COULD use (but does not MUST use).

Solving #59 means making receiving jobs of `kaniko.execute` and `crane.pull` using that default cache. So #59 still remains.